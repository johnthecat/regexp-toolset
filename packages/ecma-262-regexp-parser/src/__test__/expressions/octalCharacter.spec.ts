import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

describe('Octal char', () => {
  it('should correctly octal value', () => {
    expect(parseRegexp(/\141/)).toMatchSnapshot();
  });

  it('should fallback to char sequence, if octal value is out of range', () => {
    // Largest octal value - 377₈ (255₁₀)
    expect(parseRegexp(/\378/)).toMatchSnapshot();
  });

  it('should fallback to char sequence, if there is not enough digits', () => {
    expect(parseRegexp(/\37d/)).toMatchSnapshot('with char at the end');
    expect(parseRegexp(/\37/)).toMatchSnapshot('with regexp end at the end');
  });
});
