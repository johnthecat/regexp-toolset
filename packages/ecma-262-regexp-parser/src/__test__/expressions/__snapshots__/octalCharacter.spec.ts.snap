// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Octal char > should correctly parse octal value 1`] = `
{
  "body": {
    "charCode": 97,
    "end": 4,
    "kind": 6,
    "start": 1,
    "type": "octal",
    "value": "a",
  },
  "end": 5,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if octal value is out of range 1`] = `
{
  "body": {
    "end": 4,
    "expressions": [
      {
        "charCode": 51,
        "end": 2,
        "kind": 6,
        "start": 1,
        "type": "escaped",
        "value": "3",
      },
      {
        "charCode": 55,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "7",
      },
      {
        "charCode": 56,
        "end": 4,
        "kind": 6,
        "start": 4,
        "type": "simple",
        "value": "8",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 5,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough digits > with char at the end 1`] = `
{
  "body": {
    "charCode": 31,
    "end": 4,
    "kind": 6,
    "start": 1,
    "type": "octal",
    "value": "",
  },
  "end": 5,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`Octal char > should fallback to char sequence, if there is not enough digits > with regexp end at the end 1`] = `
{
  "body": {
    "end": 3,
    "expressions": [
      {
        "charCode": 51,
        "end": 2,
        "kind": 6,
        "start": 1,
        "type": "escaped",
        "value": "3",
      },
      {
        "charCode": 55,
        "end": 3,
        "kind": 6,
        "start": 3,
        "type": "simple",
        "value": "7",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
